packages/uniforms/src/changedKeys.ts

```diff
@@ -4,7 +4,7 @@ import { joinName } from './joinName';
 
 // eslint-disable-next-line complexity
 export function changedKeys<T>(root: string, valueA?: T, valueB?: T) {
-  if (!valueA || valueA !== Object(valueA) || valueA instanceof Date) {
+  if (!valueA || typeof valueA !== typeof valueB || valueA !== Object(valueA) || valueA instanceof Date) {
```

Comment:
As it's going to be the most uncommon case, I'd move it to the end of this condition:
```suggestion
  if (!valueA || valueA !== Object(valueA) || valueA instanceof Date || typeof valueA !== typeof valueB) {
```